### ISSUE TITLE
Session not saving

### ISSUE THREAD

**Issue by dalisoft** [expressjs/session#663]
# Title: Session not saving

## Description
Hi everyone.
I'm trying make it work already almost 5 hour, more can't.

### Type
- [ ] Question
- [ ] Discussion
- [x] Bug

### OS
- [ ] Windows
- [ ] Linux
- [x] macOS (v10.13.6)

### Node
- [x] v10
- [x] v11
- [x] v12 (`npm v6.9.0`)

### Log
```bash
[nodemon] starting `node examples/passport.js`
[Server]: started successfully at [localhost:4000] in [20ms]
  express-session cookie unsigned +0ms
  express-session cookie unsigned +0ms
  express-session no SID sent, generating session +1ms
  express-session saving hvZ29sVVqcM7rNOJIZVVuG5DMKZ2Ozr6 +4ms
  express-session cookie unsigned +3m
  express-session cookie unsigned +0ms
  express-session no SID sent, generating session +0ms
  express-session saving uD6W266EvlXI7rj5-VIbFvQPOQFNIeCT +1ms
  express-session cookie unsigned +327ms
  express-session cookie unsigned +1ms
  express-session no SID sent, generating session +0ms
  express-session saving DEXzBwc4PdH6kXCvr8wn9TbFXkrsTfTY +0ms
```

Everytime it generates new session and old variables isn't saving.

### Configuration
```js
app.use(
  expressSession({
    secret: 'session_secret',
    name: 'passport.sid',
    resave: true,
    rolling: true,
    saveUninitialized: true,
    cookie: {
      sameOrigin: false,
      secure: false,
      expires: Date.now() + 60 * 1000
    },
    logErrors: true
  })
);
```

### Example
```js
app.get('/', async (req, res) => {
  req.session.tmp = new Date().toString();
  return req.user || 'Unauthorized';
});
```

### Tried methods
I tried (and none of them worked):
- [x] include `cors` and `credentials` at client
- [x] using postman/paw
- [x] changing settings
- [x] cors setting at server

### Note
Is am i doing something wrong?
Pls help me

**Comment by conman0552** [comment]
I am experiencing this issue as well. I've scoured the internet in search of something to help me, but found nothing to help.

I'm only generating a session if a client logs in, but I've been printing logs as well. 

When a client logs in on my app, the console says:

> { path: '/',
     _expires: 2020-06-08T15:30:17.856Z,
     originalMaxAge: 31536000000,
     httpOnly: true,
     secure: false,
     domain: 'my.domain.here.com/thisisn'tmyrealdomainIamusing' },
  userid: '81482246',
  sysrank: '175',
  fear_rank: '190',
  advancement_rank: '0',
  mp_rank: '20',
  justice_rank: '3',
  vanguard_rank: '50',
  rf_rank: '0' }
[6/9/2019 11:30:17 AM] Login from (::ffff:) to conmmander OK;

However, once I log in my app will redirect to the homepage which should give me access to some secure pages. The console is now printing: 

> Session {
  cookie:
   { path: '/',
     _expires: 2020-06-08T15:37:16.002Z,
     originalMaxAge: 31536000000,
     httpOnly: true,
     secure: false,
     domain: 'my.domain.here.com/thisisn'tmyrealdomainIamusing' } }

The above logs are from doing `console.log(req.session)`
What would be causing this to happen? My configuration is as so:
`
let baseurl = "my.domain.here.com/thisisn'tmyrealdomainIamusing"
app.use(session({secret: "secretCode", name: "SessionData", resave: false, saveUninitialized: false, cookie: { maxAge: 365 * 24 * 60 * 60 * 1000, secure: false, httpOnly: true, domain: baseurl}}));`

When a client logs in, I am setting session data like so:
`req.session.userid = row['userid']
req.session.sysrank = row['sysrank']
req.session.fear_rank = fearRankNum
req.session.advancement_rank = advancementRankNum
req.session.mp_rank = mpRankNum
req.session.justice_rank = justiceRankNum
req.session.vanguard_rank =vanguardRankNum
req.session.rf_rank = rfRankNum
req.session.save()`

Figured this may be the best place to ask, since it is the most similar to my issue.

Thank you for any help!
